kvx/src/implementations/memory.rs

```diff
@@ -11,26 +12,185 @@ use crate::{
     Error, Key, KeyValueStoreBackend, ReadStore, Result, Scope, TransactionCallback, WriteStore,
 };
 
-type MemoryStore = HashMap<Key, serde_json::Value>;
+#[derive(Debug)]
+pub struct MemoryStore(HashMap<NamespaceBuf, HashMap<Key, serde_json::Value>>);
+
+impl MemoryStore {
+    fn new() -> Self {
+        MemoryStore(HashMap::new())
+    }
+
+    fn has(&self, namespace: &NamespaceBuf, key: &Key) -> bool {
+        self.0
+            .get(namespace)
+            .map(|m| m.contains_key(key))
+            .unwrap_or_default()
```

Comment:
I’d probably do an explicit `unwrap_or(false)` in these cases to make it clear what you want.

Reply:
done